### PR TITLE
Add WinUSB Manager and Rez TranceVibrator support

### PR DESCRIPTION
**Pull request by qdot**

### CODE REVIEW

**Comment by qdot** [comment]
Wow. Force pushed before I commented on the review and the review UI got weird. Wasn't sure how to actually address review comments so had to dismiss review.

File: Buttplug.Server.Managers.WinUSBManager/RezTranceVibratorDevice.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        {
+            if (!(aMsg is VibrateCmd cmdMsg))
+            {
+                return BpLogger.LogErrorMsg(aMsg.Id, Error.ErrorClass.ERROR_DEVICE, "Wrong Handler");
+            }
+
+            if (cmdMsg.Speeds.Count == 0 || cmdMsg.Speeds.Count > _vibratorCount)
+            {
+                return new Error(
+                    _vibratorCount == 1 ? "VibrateCmd requires 1 vector for this device." :
```
Comment (blackspherefollower) [inline Buttplug.Server.Managers.WinUSBManager/RezTranceVibratorDevice.cs]: This is never going to be more than 1 right?

File: Buttplug.Server.Managers.WinUSBManager/RezTranceVibratorDevice.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        }
+
+        private Task<ButtplugMessage> HandleStopDeviceCmd(ButtplugDeviceMessage aMsg)
+        {
+            BpLogger.Debug("Stopping Device " + Name);
+            return HandleSingleMotorVibrateCmd(new SingleMotorVibrateCmd(aMsg.DeviceIndex, 0, aMsg.Id));
+        }
+
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+        private async Task<ButtplugMessage> HandleVibrateCmd(ButtplugDeviceMessage aMsg)
```
Comment (blackspherefollower) [inline Buttplug.Server.Managers.WinUSBManager/RezTranceVibratorDevice.cs]: If you return the responses like `return Task.FromResult<ButtplugMessage>(new Ok(aMsg.Id));` you can remove the async modifier.

File: Buttplug.Server.Managers.WinUSBManager/RezTranceVibratorDevice.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            return new Ok(aMsg.Id);
+        }
+
+        private async Task<ButtplugMessage> HandleSingleMotorVibrateCmd(ButtplugDeviceMessage aMsg)
+        {
+            var cmdMsg = aMsg as SingleMotorVibrateCmd;
+            if (cmdMsg is null)
+            {
+                return BpLogger.LogErrorMsg(aMsg.Id, Error.ErrorClass.ERROR_DEVICE, "Wrong Handler");
+            }
```
Comment (blackspherefollower) [inline Buttplug.Server.Managers.WinUSBManager/RezTranceVibratorDevice.cs]: Is reshaper not turning the code directly after closed brackets into syntax errors? Of is it fxcop that does that? Either way, looks like linting isn't the same here as elsewhere.

File: Buttplug.Server.Managers.WinUSBManager/Buttplug.Server.Managers.WinUSBManager.csproj [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  <ItemGroup>
+    <ProjectReference Include="..\Buttplug.Core\Buttplug.Core.csproj">
+      <Project>{fc4af8fe-8c21-4e07-8d24-7fb7791149d0}</Project>
+      <Name>Buttplug.Core</Name>
+    </ProjectReference>
+    <ProjectReference Include="..\Buttplug.Server\Buttplug.Server.csproj">
+      <Project>{23d9ec18-2dba-4149-a951-b84786d4bfa6}</Project>
+      <Name>Buttplug.Server</Name>
+    </ProjectReference>
+  </ItemGroup>
```
Comment (blackspherefollower) [inline Buttplug.Server.Managers.WinUSBManager/Buttplug.Server.Managers.WinUSBManager.csproj]: I think this is missing:
```
  <ItemGroup>
    <AdditionalFiles Include="..\Buttplug.ProjectFiles\stylecop.json" />
  </ItemGroup>
  <ItemGroup>
    <Analyzer Include="..\packages\StyleCop.Analyzers.1.1.0-beta004\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll" />
    <Analyzer Include="..\packages\StyleCop.Analyzers.1.1.0-beta004\analyzers\dotnet\cs\StyleCop.Analyzers.dll" />
  </ItemGroup>
```

File: Buttplug.Server.Managers.WinUSBManager/packages.config [hunk 5]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="WinUSBNet" version="1.0.3" targetFramework="net452" />
```
Comment (blackspherefollower) [inline Buttplug.Server.Managers.WinUSBManager/packages.config]: Add ```<package id="StyleCop.Analyzers" version="1.1.0-beta004" targetFramework="net452" developmentDependency="true" />```

File: Buttplug.Server.Managers.WinUSBManager/RezTranceVibratorDevice.cs [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+using System.Threading.Tasks;
+using Buttplug.Core;
+using Buttplug.Core.Messages;
+using MadWizard.WinUSBNet;
+
+namespace Buttplug.Server.Managers.WinUSBManager
+{
+    internal class RezTranceVibratorDevice : ButtplugDevice
+    {
+        private static uint _vibratorCount = 1;
```
Comment (blackspherefollower) [inline Buttplug.Server.Managers.WinUSBManager/RezTranceVibratorDevice.cs]: You could probably get rid of this, since it never varies, but not a big deal

**Pull request merged**